### PR TITLE
Payment method specific fields

### LINKED ISSUE #134: Payment specific parameters for payment service
**Describe the solution you'd like**
Right now it is only available to specify payment specific parameters for orders even though Mollie supports this for payments as well. Right now its impossible to create an ideal payment and specifying the issue at the same time using the Payments service while Mollie does support this. The order implementation does support payment specific parameters but I'd like to use the payment API.

**Describe alternatives you've considered**
Manual requests are the only way to achieve this right now which is not the best way to work with this library.

### PR DESCRIPTION
**Pull request by VictorAvelar**
## Description

When dealing with create payment requests it was not possible to specify some payment method specific values, now that the fields where added and given the fact that the values are optional for Mollie it should be possible to both create `core` payment and also `enriched` payments

## Motivation and context

Fixes #134 

## How has this been tested?

Specific test cases added to the PR.

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

Go over all the following points, and put an `x` in all the boxes that apply.

Please, please, please, don't send your pull request until all of the boxes are ticked. Once your pull request is created, it will trigger a build on our continuous integration server to make sure your [tests and code style pass](https://help.github.com/articles/about-required-status-checks/).

- [x] I have read the **[CONTRIBUTING](CONTRIBUTING.md)** document.
- [x] My pull request addresses exactly one patch/feature.
- [x] I have created a branch for this patch/feature.
- [x] Each individual commit in the pull request is meaningful.
- [x] I have added tests to cover my changes.
- [x] If my change requires a change to the documentation, I have updated it accordingly.

If you're unsure about any of these, don't hesitate to ask. We're here to help!

### CODE REVIEW

**Comment by VictorAvelar** [comment]
@meruiden it is ready if you wanna have a look, I will probably merge and tag it tonight.

File: mollie/payments.go [hunk 1]
```diff
@@ -43,44 +43,60 @@ const (
...
+	Mode                            Mode            `json:"mode,omitempty"`
+	Locale                          Locale          `json:"locale,omitempty"`
 	RestrictPaymentMethodsToCountry Locale          `json:"restrictPaymentMethodsToCountry,omitempty"`
-	SubscriptionID                  string          `json:"subscriptionId,omitempty"`
-	TestMode                        bool            `json:"testmode,omitempty"`
+	Method                          PaymentMethod   `json:"method,omitempty"`
+	Links                           PaymentLinks    `json:"_links,omitempty"`
+	SequenceType                    SequenceType    `json:"sequenceType,omitempty"`
+	// Payment method specific params.
+	DigitalGoods         bool                   `json:"digitalGoods,omitempty"`
```
Comment (meruiden) [inline mollie/payments.go:85]: There is some inconsistency here with the indentation compared to the original parameters.

**Review by meruiden — CHANGES_REQUESTED** [review]
Everything looks perfect except for some indentation inconsistency

File: mollie/payments.go (hunk 1 above)
Comment (VictorAvelar) [reply mollie/payments.go:85]: yeah, it is because of the comment in the middle of the struct declaration, I will fix it and then merge

**Pull request merged**